File: fams-bo/resource/com/purefun/fams/core/bo/otw/TestBO_OTW.java
package com.purefun.fams.core.bo.otw;

import com.purefun.fams.core.bo.TestBO;
import com.purefun.fams.core.bo.pro.TestBO_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fams.core.bo.commom.ICommon_OTW;
import com.google.protobuf.Any;

public class TestBO_OTW implements ICommon_OTW {
    TestBO_PRO.TestBO.Builder builder = null;
    TestBO bo = null;

    public TestBO_OTW() {
        builder = TestBO_PRO.TestBO.newBuilder();
        bo= new TestBO();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public TestBO_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = TestBO_PRO.TestBO.newBuilder();
        bo= new TestBO();
        TestBO_PRO.TestBO receive = TestBO_PRO.TestBO.parseFrom(message);
        setUsername(receive.getUsername());
        setAge(receive.getAge());
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
    }

    public TestBO_OTW(TestBO bofrom){
        builder = TestBO_PRO.TestBO.newBuilder();
        bo= new TestBO();
        setUsername(bofrom.username);
        setAge(bofrom.age);
        setUuid(bofrom.uuid);
        setBoid(bofrom.boid);
        setDestination(bofrom.destination);
    }

    public byte[] serial() {
        return builder.build().toByteArray();
    }

    @Override
    public com.google.protobuf.GeneratedMessageV3.Builder getBuilder() { 
        return builder;
    }

    @Override
    public TestBO getBo() { 
        return bo;
    }

    public java.lang.String getUsername() {
        return builder.getUsername();
    }

    public void setUsername(java.lang.String username) {
        bo.username = username;
        builder.setUsername(username);
    }

    public int getAge() {
        return builder.getAge();
    }

    public void setAge(int age) {
        bo.age = age;
        builder.setAge(age);
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public String toString() {
        return "TestBO_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "username = " + getUsername() +"," +
            "age = " + getAge() +"," +
         "]";
    }
}

File: fams-python-service/src/com/purefun/fams/core/bo/otw/TestBO_OTW.py
from src.core.common.ICommon_OTW import ICommon_OTW
from src.com.purefun.fams.core.bo.pro import TestBO_pb2
from src.com.purefun.fams.core.bo.model.TestBO import TestBO

class TestBO_OTW(ICommon_OTW):

    def __init__(self, byteMsg = None):
        self._bo_pro = TestBO_pb2.TestBO()
        self._bo = TestBO()

        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()

    def __setDataFromBO(self):
        self._bo_pro.username = self._bo.username
        self._bo_pro.age = self._bo.age
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.destination = self._bo.destination

    def __setDataFromPB(self):
        self._bo.username = self._bo_pro.username
        self._bo.age = self._bo_pro.age
        self._bo.uuid = self._bo_pro.uuid
        self._bo.boid = self._bo_pro.boid
        self._bo.destination = self._bo_pro.destination

    def getBO(self):
        return self._bo

    def getProBO(self):
        return self._bo_pro

    def getUsername(self):
        return self._bo.username

    def setUsername(self, username):
        self._bo.username = username
        self._bo_pro.username = username

    def getAge(self):
        return self._bo.age

    def setAge(self, age):
        self._bo.age = age
        self._bo_pro.age = age

    def getUuid(self):
        return self._bo.uuid

    def setUuid(self, uuid):
        self._bo.uuid = uuid
        self._bo_pro.uuid = uuid

    def getBoid(self):
        return self._bo.boid

    def setBoid(self, boid):
        self._bo.boid = boid
        self._bo_pro.boid = boid

    def getDestination(self):
        return self._bo.destination

    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination

    def toString(self):
        return "TestBO_OTW ["+"username = " + self.getUsername() +"," +"age = " + str(self.getAge()) +"," +"uuid = " + self.getUuid() +"," +"boid = " + str(self.getBoid()) +"," +"destination = " + self.getDestination() +"," +"]"
